### PR TITLE
[media] Report errors via MediaError to WebApp

### PR DESCRIPTION
**Pull request by borongc**
This PR adds `StarboardRendererFactory`, which contains `media_log_` to `StarboardRenderer`, and is necessary for error reporting.

Chromium uses both `RendererClient::OnError` and `media_log_` to report error code and message to `MediaError`. As a result, Cobalt adapts it from C25 to report error messages to WebApp.

b/375271948
b/375278384

### CODE REVIEW

**Review by jasonzhangxx — APPROVED** [review]

**Comment by borongc** [comment]
Note that part of implementation of `StarboardRendererFactory` is taken from https://github.com/youtube/cobalt/pull/4407.

File: media/mojo/mojom/BUILD.gn
Comment (yell0wd0g) [inline media/mojo/mojom/BUILD.gn:130]: Pending GN review, I think this can be just `if (use_starboard_media) {`, and above doesnt' need `if (is_cobalt)`. Same throughout.

File: media/starboard/BUILD.gn
Comment (yell0wd0g) [inline media/starboard/BUILD.gn:1]: Unrelated but I'd change l.15
```
assert(is_cobalt, "This file builds for cobalt builds, not Chromium builds")
```

with 

```
assert(is_starboard_media)
```

File: media/starboard/starboard_renderer.cc
Comment (yell0wd0g) [inline media/starboard/starboard_renderer.cc:741]: Would this `kSbPlayerErrorMax` ever appear?

File: media/starboard/starboard_renderer.cc
Comment (yell0wd0g) [inline media/starboard/starboard_renderer.cc:726]: Should we have a fallback for other cases not specified now?

File: media/starboard/starboard_renderer.h
Comment (yell0wd0g) [inline media/starboard/starboard_renderer.h:1]: Shouldn't we have a unittest file like https://source.chromium.org/chromium/chromium/src/+/main:media/renderers/renderer_impl_unittest.cc ?

File: third_party/blink/renderer/platform/media/web_media_player_impl.cc
Comment (yell0wd0g) [inline third_party/blink/renderer/platform/media/web_media_player_impl.cc:2827]: This is already logged in Media DevTools in l.2824, is this extra LOG() needed?

**Review by yell0wd0g — CHANGES_REQUESTED** [review]
Overall looking good.

**Comment by yell0wd0g** [comment]
@niranjanyardi @andrewsavage1 for the questions on GN files.

File: media/starboard/BUILD.gn
Comment (niranjanyardi) [reply media/starboard/BUILD.gn:1]: My understanding with a previous chat with Borong is this file may compile when `use_starboard_media = false` in the future : we want to compare cobalt with starboard media, cobalt without starboard media on android. I'll leave it to Borong to add the assert if needed.

**Review by niranjanyardi — APPROVED** [review]
Apart from the open comments lgtm from a build perspective.

File: media/mojo/mojom/BUILD.gn
Comment (andrewsavage1) [reply media/mojo/mojom/BUILD.gn:130]: This is done correctly as is, see go/cobalt-naming#gn for an example. I can expand more on that section if needed

Comment (niranjanyardi) [reply media/mojo/mojom/BUILD.gn:130]: Hmm, that's interesting , looking at go/cobalt-naming#values-for-common-variables `is_cobalt` seems redundant in this case. 

I do see we have a naming system of (is_cobalt && use_foo) - I would argue for removing the redundant flag when needed. But that's a separate discussion - shouldn't block this PR

File: content/renderer/media/media_factory.cc
Comment (borongc) [reply content/renderer/media/media_factory.cc:749]: Done.

File: media/starboard/BUILD.gn
Comment (borongc) [reply media/starboard/BUILD.gn:1]: The BUILD.gn file is built when `is_cobalt=true`. `StarboardRenderer` is a subcomponent for `//media` (https://github.com/youtube/cobalt/blob/8125775568692d6f40cdfae22d014c6d39e4f059/media/media_options.gni#L354), unless we change it to `use_starboard_media`, it is right here.

File: media/starboard/starboard_renderer.cc
Comment (borongc) [reply media/starboard/starboard_renderer.cc:726]: For SbPlayer, we only have `kSbPlayerErrorDecode` and `kSbPlayerErrorCapabilityChanged` for errors. It should be fine here for not specifying a default case.

File: media/starboard/starboard_renderer.cc
Comment (borongc) [reply media/starboard/starboard_renderer.cc:741]: I don't think so.

File: media/starboard/starboard_renderer.h
Comment (borongc) [reply media/starboard/starboard_renderer.h:1]: We can add unit-tests for this after we enable tests for Cobalt.

File: third_party/blink/renderer/platform/media/web_media_player_impl.cc
Comment (borongc) [reply third_party/blink/renderer/platform/media/web_media_player_impl.cc:2827]: We may not always use DevTools for media. This can ensure we can have this in adb bugreport.

File: media/starboard/starboard_renderer.h
Comment (borongc) [reply media/starboard/starboard_renderer.h:1]: Created a bug (b/379927913) for adding unit-tests for StarboardRenderer.

File: media/mojo/mojom/BUILD.gn
Comment (andrewsavage1) [reply media/mojo/mojom/BUILD.gn:130]: `is_cobalt` is not redundant, as if `is_cobalt = false`, `use_starboard_media` is undefined.

File: media/starboard/starboard_renderer.cc
Comment (yell0wd0g) [reply media/starboard/starboard_renderer.cc:726]: Gotcha!

File: media/starboard/starboard_renderer_factory.h
Comment (yell0wd0g) [inline media/starboard/starboard_renderer_factory.h:28]: nit: `explicit`.

**Review by yell0wd0g — APPROVED** [review]
LGTM !

**Pull request merged**
